qt/scientific_interfaces/ISISReflectometry/README.md

```diff
@@ -0,0 +1,90 @@
+
+### `Reduction`
+
+This contains models which centre around the "reduction configuration". This is a representation of all of the data that has been entered in the GUI. The top level `Batch` model provides everything needed to perform a reduction.
+
+Additionally these models also contain some state information, e.g. the `Row` and `Group` contain information about whether processing has been performed
+
+### `Common`
+
+Contains non-GUI-specific utility files useful in more that one component of the reflectometry interface but that are still specific to this interface; more generic utilities should be put elsewhere, e.g. in Framework.
+
+### `TestHelpers`
+
+This file contains components specific for unit testing. The actual tests are in `../test/ISISReflectometry/`.
+
+## `Reduction back-end`
+
+The back-end is primarily a set of algorithms, with the entry points from the GUI being ReflectometryISISLoadAndProcess (for reducing a row) and Stitch1DMany (for post-processing a group). Any additional processing should be added to these algorithms, or a new wrapper algorithm could be added if appropriate (this might be necessary in future if post-processing will involve more than just stitching).
+
+The `BatchPresenter` is the main coordinator for executing a reduction. The `BatchJobRunner` converts the reduction configuration to a format appropriate for the algorithms. The conversion functions are in `RowProcessingAlgorithm` and `GroupProcessingAlgorithm`, and any algorithm-specific code should be kept to these files.
+
+Unfortunately the whole batch cannot be farmed off to a single algorithm because we need to update the GUI after each row completes, and we must be able to interrupt processing so that we can cancel a large batch operation. We also need to know whether rows completed successfully before we can set up the group post-processing algorithms. Some queue management is therefore done by the `BatchPresenter`, with the help of the `BatchAlgorithmRunner`.
+
+## `Development Guidelines`
+
+The following design principles should be adhered to when developing the GUI. If current design does not seem appropriate for additional feature requests, do consult with a senior developer to work out the best way forward rather than proceeding in a non-optimal way.
+
+### Adhere to MVP
+
+To ensure the GUI can be easily tested we follow the MVP design pattern. There is general guidance on this here: https://developer.mantidproject.org/GUIDesignGuidelines.html
+
+The view cannot be tested so should be kept as simple as possible - typically any user action on the view results in a notification to the presenter and is handled from there (even if that is just an update back to the view).
+
+The views should not have a direct pointer to their presenters, so the notification is done via a subscriber interface. The only exception is the QtMainWindowView, but notifications should still be done via the subscriber interface.
```

Comment:
Could we give one example of the subscriber interface, so someone reading the doc could go and look how the pattern is used in practice?